pkg/planner/funcdep/fd_graph.go

```diff
@@ -1224,3 +1224,52 @@ func (s *FDSet) IsHashCodeRegistered(hashCode string) (int, bool) {
 	}
 	return -1, false
 }
+
+// FindCommonEquivClasses find the common equivalences between multi fds.
+func FindCommonEquivClasses(fdSets []*FDSet) []intset.FastIntSet {
+	if len(fdSets) == 0 {
+		return nil
+	}
+	var result []intset.FastIntSet
+	for _, edge := range fdSets[0].fdEdges {
+		if edge.equiv {
+			// clone one for later mutable intersection.
+			set := intset.NewFastIntSet()
+			set.CopyFrom(edge.from)
+			result = append(result, set)
+		}
+	}
+	// iterate intersection with later all fdSets.
+	for i := 1; i < len(fdSets); i++ {
+		var newResult []intset.FastIntSet
```

Comment:
Although we cannot see right capacity for it. we can prealloc small size memory for it to avoid slicesgrow.